### PR TITLE
Add jobs for reindexing from RabbitMQ messages

### LINKED ISSUE #728: Trigger indexing when a RabbitMQ message is created
Currently indexing depends on Fedora 3 sending ActiveMQ messages, which are routed by Karaf to dor_indexing_app.  We will replace this functionality by enabling dor_indexing_app to listen to RabbitMQ messages. (Similar to how h2 does) and turn off the Karaf route that sends ActiveMQ messages to dor_indexing_app

DSA already sends messages: https://github.com/sul-dlss/dor-services-app/blob/e8d7595f96c62b8b4a7c13a98618c2c66a521504/app/services/notifications/object_updated.rb

Workflow-server-rails already sends messages https://github.com/sul-dlss/workflow-server-rails/blob/e328f768446aa37dcd067d830f9833c7c4f36c57/app/services/send_update_message.rb#L9

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #728

### CODE REVIEW

File: app/jobs/reindex_by_druid_job.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Reindexes an object given a druid
+class ReindexByDruidJob
+  include Sneakers::Worker
+  # This worker will connect to "dor.indexing-with-model" queue
+  # env is set to nil since by default the actual queue name would be
+  # "dor.indexing-with-model_development"
+  from_queue 'dor.indexing-by-druid', env: nil
```
Comment (mjgiarlo) [inline app/jobs/reindex_by_druid_job.rb:9]: Should this comment block have `s/with-model/by-druid/g`?

File: config/settings.yml [hunk 2]
```diff
@@ -20,3 +20,10 @@ workflow_url: 'https://workflow.example.edu/'
 dor_services:
   url: 'http://localhost:3003'
   token: 'not-a-real-token'
+
+
+rabbitmq:
+  hostname: localhost
+  vhost: /
+  username: guest
+  password: guest
```
Comment (mjgiarlo) [inline config/settings.yml:29]: Looks good. I see that QA and Stage in shared_configs are configured slightly differently. Looks like one has a vhost and one doesn't. Intended? And I assume doing this for prod is in your plan later on.

**Review by mjgiarlo — APPROVED** [review]
I have a minor suggestion re: a code comment but am approving

File: config/settings.yml (hunk 2 above)
Comment (jcoyne) [reply config/settings.yml:29]: It's a PR awaiting review: https://github.com/sul-dlss/shared_configs/pull/1723

File: app/jobs/reindex_by_druid_job.rb (hunk 1 above)
Comment (jcoyne) [reply app/jobs/reindex_by_druid_job.rb]: Yep, I've made this fix.

**Pull request merged**